packages/application-extension/src/index.tsx

```diff
@@ -429,38 +449,196 @@ const sidebar: JupyterFrontEndPlugin<void> = {
 /**
  * Add the main application commands.
  */
-function addCommands(app: JupyterLab, palette: ICommandPalette): void {
+function addCommands(
+  app: JupyterLab,
+  palette: ICommandPalette,
+  labShell: ILabShell,
+  docManager: IDocumentManager
+): void {
+  const { commands, contextMenu, shell } = app;
   const category = 'Main Area';
-  let command = CommandIDs.activateNextTab;
 
-  app.commands.addCommand(command, {
+  // Returns the doc widget associated with the most recent contextmenu event.
+  const contextMenuWidget = (): Widget => {
```

Comment:
Yikes, this actually does rely on the document manager! In which case, I think it might not be right to move this stuff to the application.

Reply:
I updated so that none of these commands depend on doc manager

Reply:
It still implicitly depends on the docmanager: the path regex in this function uses the hover text that the doc manager adds to document widgets. I think we may want to discuss a bit further exactly what this `contextMenuWidget` should do.